mmdet/apis/test.py

```diff
@@ -8,19 +8,36 @@
 import torch.distributed as dist
 from mmcv.runner import get_dist_info
 
+from mmdet.core import tensor2imgs
 
-def single_gpu_test(model, data_loader, show=False):
+
+def single_gpu_test(model, data_loader, show=False, out_dir=None):
     model.eval()
     results = []
     dataset = data_loader.dataset
     prog_bar = mmcv.ProgressBar(len(dataset))
     for i, data in enumerate(data_loader):
         with torch.no_grad():
-            result = model(return_loss=False, rescale=not show, **data)
+            result = model(return_loss=False, rescale=True, **data)
         results.append(result)
 
-        if show:
-            model.module.show_result(data, result)
+        if show or out_dir:
+            img_tensor = data['img'][0]
+            img_metas = data['img_metas'][0].data[0]
+            imgs = tensor2imgs(img_tensor, **img_metas[0]['img_norm_cfg'])
+            assert len(imgs) == len(img_metas)
+
+            for img, img_meta in zip(imgs, img_metas):
+                h, w, _ = img_meta['img_shape']
+                img_show = img[:h, :w, :]
+                img_show = mmcv.imresize(img_show,
+                                         img_meta['ori_shape'][:-1][::-1])
```

Comment:
Explicitly specifying the shape may be helpful for users to understand the codes.
```python
h, w = img_meta['ori_shape'][:-1]
img_show = mmcv.imresize(img_show, (w, h))
```